calculator.rb

```diff
@@ -0,0 +1,64 @@
+class Calculator
+  def initialize(calculate_sum: Sum.new, calculate_subtraction: Subtraction.new, calculate_multiplication: Multiplication.new, calculate_division: Division.new)
+    @calculate_sum = calculate_sum
+    @calculate_subtraction = calculate_subtraction
+    @calculate_multiplication = calculate_multiplication
+    @calculate_division = calculate_division
+  end
```

Comment:
Uma forma de fazer a injeção das nossas dependências sem o método fique com muitos parâmetros é usando um hash que terá as instâncias do que queremos injetar e se ele não tiver aquela chave a gente cria uma instância dentro do próprio construtor da classe. Fica mais ou menos assim:
```suggestion
  def initialize(overrides = {})
    @calculate_sum = overrides.fetch(:calculate_sum) do
      Sum.new
    end
  end
```